providers/parseable/src/lib.rs

```diff
@@ -0,0 +1,178 @@
+    let config = Config::parse(request.config)?;
+    let url = get_url(LIVENESS_API, &config)?;
+    let response = make_http_request(HttpRequest::get(url)).await?;
+    if response.status_code == 200 {
+        ProviderStatus::builder()
+            .status(Ok(()))
+            .version(COMMIT_HASH.to_owned())
+            .built_at(BUILD_TIMESTAMP.to_owned())
+            .build()
+            .to_blob()
+    } else {
+        Err(Error::Http {
+            error: HttpRequestError::Offline,
+        })
+    }
+}
+
+async fn run_query(query: &Query, config: &Config) -> Result<Vec<ProviderEvent>> {
+    let query = json!({
+        "query": query.query,
+        "startTime": query.time_range.from.format(&well_known::Rfc3339).unwrap(),
+        "endTime": query.time_range.to.format(&well_known::Rfc3339).unwrap()
+    });
+
+    let url = get_url(QUERY_API, config)?;
+    let mut headers = BTreeMap::new();
+    headers.insert(
+        "Authorization".to_string(),
+        basic_auth_header(&config.username, &config.password),
+    );
+    headers.insert("Content-Type".to_string(), "application/json".to_string());
+
+    let request = HttpRequest::post(url, query.to_string()).with_headers(headers);
+    let response = make_http_request(request).await?;
+
+    let body: Value = serde_json::from_slice(&response.body)?;
+    let Value::Array(arr) = body  else {
+        return Err(Error::Other { message: String::from_utf8_lossy(&response.body).to_string() })
+    };
+
+    let mut rows = Vec::new();
+    for value in arr {
+        rows.push(parse_row(value))
+    }
+
+    Ok(rows)
+}
+
+fn parse_row(value: Value) -> ProviderEvent {
+    let Value::Object(mut object) = value  else { panic!("object is of type {:?}", &value) };
+
+    let timestamp = object.remove("p_timestamp").unwrap();
+    let timestamp = Timestamp::from(parse_time(timestamp.as_str().unwrap()));
+
+    let otel = OtelMetadata::builder()
+        .attributes(BTreeMap::from_iter(object))
+        .resource(BTreeMap::default())
+        .build();
+
+    ProviderEvent::builder()
+        .otel(otel)
+        .time(timestamp)
+        .title("".to_string())
+        .build()
+}
+
+fn get_url(api: &str, config: &Config) -> Result<Url> {
+    let base_url: Url = config
+        .endpoint
+        .parse()
+        .map_err(|e: url::ParseError| Error::Config {
+            message: format!("Invalid URL in configuration: {e:?}"),
+        })?;
+    base_url.join(api).map_err(|e| Error::Config {
+        message: format!("Invalid URL in configuration: {e:?}"),
+    })
+}
+
+fn basic_auth_header(username: &str, password: &str) -> String {
+    format!(
+        "Basic {}",
+        BASE64_STANDARD.encode(&format!("{}:{}", username, password))
+    )
+}
+
+// Parseable's p_timestamp value is not sufficient for time parser.
+// Offset is missing though it is understood as UTC. So we add it manually
+fn parse_time(time: &str) -> OffsetDateTime {
```

Comment:
I think this one would also be better if it returned a `Result<OffsetDateTime>` (or directly `Result<Timestamp>`).